File: Arrays_&_Hashing/Valid_Anagram/1_BruteForce.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        boolean[] visited = new boolean[t.length()];
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            boolean found = false;
            for (int j = 0; j < t.length(); j++) {
                if (!visited[j] && t.charAt(j) == c) {
                    visited[j] = true;
                    found = true;
                    break;
                }
            }
            if (!found) return false;
        }
        return true;
    }
}

File: Arrays_&_Hashing/Valid_Anagram/1_BruteForce.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        vector<bool> visited(t.size(), false);
        for (char c : s) {
            bool found = false;
            for (size_t i = 0; i < t.size(); ++i) {
                if (!visited[i] && t[i] == c) {
                    visited[i] = true;
                    found = true;
                    break;
                }
            }
            if (!found) return false;
        }
        return true;
    }
};
